src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs

```diff
@@ -370,14 +372,42 @@ public Utf8String Substring(int index, int length)
             }
         }
 
-        // TODO: Should this be public or should focused on scenarios?
-        private int IndexOf(Utf8String value)
+        // TODO: Naive algorithm, reimplement faster
+        // TODO: Should this be public?
+        public int IndexOf(Utf8String value)
         {
-            throw new NotImplementedException();
+            if (value.Length == 0)
+            {
+                // TODO: Is this the right answer?
+                // TODO: Does this even make sense?
```

Comment:
I think it's fine to return 0 in the case where *this* instance was not empty, but I think it would be better if the return a value that can be used as an index into the *this" instance if the value is not StringNotFound, i.e. "".IndexOf("") should return StringNotFound.

I realize that this is not consistent with System.String, but Utf8String is not consistent with system.string anyway in many more significant ways. thoughts?